ui/apps/platform/src/Containers/Vulnerabilities/WorkloadCves/Overview/WorkloadCvesOverviewPage.tsx

```diff
@@ -140,6 +142,10 @@ const searchFilterConfig = {
 function WorkloadCvesOverviewPage() {
     const apolloClient = useApolloClient();
 
+    const [defaultSearchFilter, setDefaultSearchFilter] = useState<
```

Comment:
Nit: We could define this after the hook that handles the entity tab, and use the current tab as the starting value so that we can remove `undefined` from the possible values.

Reply:
That got me thinking. I think we have a few places where I had to explicitly call `setDefaultSearchFilter` like the `onChangeObservedCveMode` and `onVulnerabilityStateChange` functions. To make things future-proof, I decided to use `useEffect`. I'll watch for changes to `activeEntityTabKey` and then set the default search filter accordingly.

Reply:
Hmm in general I've been trying to avoid `useEffect` altogether whenever possible as I think almost all use-cases are somewhat of an antipattern. Do you think we could ditch both the `useEffect` and the `useState` that hold the `defaultSearchFilter` altogether, and just calculate it during each render like below? It would have the same effect (no pun intended) as syncing the two values to ensure the default filter is always dependent on the value of the entity tab.
```ts
const defaultSearchFilter = getSearchFilterEntityByTab(activeEntityTabKey);
```